docs/src/components/ComponentDoc/ComponentDoc.js

```diff
@@ -31,18 +31,31 @@ class ComponentDoc extends Component {
     sidebarSections: docTypes.sidebarSections.isRequired,
   }
 
-  state = {}
   examplesRef = createRef()
 
-  getChildContext() {
-    return {
-      onPassed: this.handleExamplePassed,
+  constructor(props) {
+    super(props)
+    const { displayName } = props
+
+    this.state = {
+      displayName,
     }
   }
 
-  componentWillReceiveProps({ displayName }) {
-    if (displayName !== this.props.displayName) {
-      this.setState({ activePath: undefined })
+  static getDerivedStateFromProps(props, state) {
+    const { displayName } = props
+
+    const derivedState = {}
+    if (displayName !== state.displayName) {
+      derivedState.displayName = displayName
+      derivedState.activePath = undefined
+    }
+    return derivedState
+  }
```

Comment:
Actually, we this can be simplified 🚀 

#### 1. We want always set `displayName` to state:
```js
return { displayName: props.displayName }
```
#### 2. We want reset only if `displayName`s are not equal:
```js
return { 
  activePath: props.displayName === state.displayName ? state.activePath : undefined,
  displayName: props.displayName,
}
```
### 3. And for this code we can remove `constructor()` at all, because at first render after `getDerivedStateFromProps()` will be called, we will reset `activePath` to undefined, but it's undefined by default 👍